src/PowerShellEditorServices/Services/PowerShell/Host/EditorServicesConsolePSHost.cs

```diff
@@ -0,0 +1,362 @@
+    {
+        private readonly ILogger _logger;
+
+        private readonly Stack<PowerShellContextFrame> _psFrameStack;
+
+        private readonly PowerShellFactory _psFactory;
+
+        private readonly ConsoleReplRunner _consoleReplRunner;
+
+        private readonly PipelineThreadExecutor _pipelineExecutor;
+
+        private readonly HostStartupInfo _hostInfo;
+
+        private readonly ReadLineProvider _readLineProvider;
+
+        private readonly Stack<KeyValuePair<Runspace, RunspaceInfo>> _runspacesInUse;
+
+        private string _localComputerName;
+
+        private int _hostStarted = 0;
+
+        public EditorServicesConsolePSHost(
+            ILoggerFactory loggerFactory,
+            ILanguageServerFacade languageServer,
+            HostStartupInfo hostInfo)
+        {
+            _logger = loggerFactory.CreateLogger<EditorServicesConsolePSHost>();
+            _psFrameStack = new Stack<PowerShellContextFrame>();
+            _psFactory = new PowerShellFactory(loggerFactory, this);
+            _runspacesInUse = new Stack<KeyValuePair<Runspace, RunspaceInfo>>();
+            _hostInfo = hostInfo;
+            Name = hostInfo.Name;
+            Version = hostInfo.Version;
+
+            _readLineProvider = new ReadLineProvider(loggerFactory);
+            _pipelineExecutor = new PipelineThreadExecutor(loggerFactory, hostInfo, this, _readLineProvider);
+            ExecutionService = new PowerShellExecutionService(loggerFactory, this, _pipelineExecutor);
+            UI = new EditorServicesConsolePSHostUserInterface(loggerFactory, _readLineProvider, hostInfo.PSHost.UI);
+
+            if (hostInfo.ConsoleReplEnabled)
+            {
+                _consoleReplRunner = new ConsoleReplRunner(loggerFactory, this, _readLineProvider, ExecutionService);
+            }
+
+            DebugContext = new PowerShellDebugContext(loggerFactory, languageServer, this,  _consoleReplRunner);
+        }
+
+        public override CultureInfo CurrentCulture => _hostInfo.PSHost.CurrentCulture;
+
+        public override CultureInfo CurrentUICulture => _hostInfo.PSHost.CurrentUICulture;
+
+        public override Guid InstanceId { get; } = Guid.NewGuid();
+
+        public override string Name { get; }
+
+        public override PSHostUserInterface UI { get; }
+
+        public override Version Version { get; }
+
+        public bool IsRunspacePushed { get; private set; }
+
+        internal bool IsRunning => _hostStarted != 0;
+
+        public Runspace Runspace => CurrentPowerShell.Runspace;
+
+        internal string InitialWorkingDirectory { get; private set; }
+
+        internal PowerShellExecutionService ExecutionService { get; }
+
+        internal PowerShellDebugContext DebugContext { get; }
+
+        internal SMA.PowerShell CurrentPowerShell => CurrentFrame.PowerShell;
+
+        internal RunspaceInfo CurrentRunspace => CurrentFrame.RunspaceInfo;
```

Comment:
```
internal RunspaceInfo CurrentRunspace => CurrentFrame?.RunspaceInfo;
```

This needs to be done in addition to the change below to prevent the null reference that would occur after returning null instead of allowing Peek() to throw an exception.